### ISSUE TITLE
[YSQL] Object is not running after restoring DB to previous state using PITR

### ISSUE THREAD

**Issue by qvad** [yugabyte/yugabyte-db#16361]
Jira Link: [DB-5778](https://yugabyte.atlassian.net/browse/DB-5778)
### Description

Started to appear at 2.14.7.0-b1, reproduced in the latest b51

```
master_gflags={
    "schedule_snapshot_rpcs_out_of_band": "true",
    "max_concurrent_snapshot_rpcs": "-1",
    "max_concurrent_snapshot_rpcs_per_tserver": "5",
    "tablet_split_low_phase_size_threshold_bytes": "2097152",
    "tablet_split_limit_per_table": "8172",
    "enable_automatic_tablet_splitting": "true",
    "tablet_split_low_phase_shard_count_per_node": "134217728",
    "ysql_num_shards_per_tserver": "1",
    "yb_num_shards_per_tserver": "1",
    'client_read_write_timeout_ms': str(30 * 60 * 1000),
    'yb_client_admin_operation_timeout_sec': str(30 * 60)
},
tserver_gflags={
    'client_read_write_timeout_ms': str(30 * 60 * 1000),
    'yb_client_admin_operation_timeout_sec': str(30 * 60),
    "enable_automatic_tablet_splitting": "true",
    "ysql_num_shards_per_tserver": "1",
    "yb_num_shards_per_tserver": "1",
}
```

Scenario is 
1. uploading data 
2. remember data state (count, max/min key value)
3. restart VM 
4. Repeat 20 times: upload more data, restore to point 2, select data state

```
"postgresqlkeyvalue0": 
"Expected {
"count": "555954", 
"max_key": "f89dbbfe-ada4-408a-a0fe-0dfd27869022:99990", 
"min_key": "f89dbbfe-ada4-408a-a0fe-0dfd27869022:0", 

 actual: 
"count": "ERROR:  Service unavailable: The object \"000033e6000030008000000000000000.postgresqlkeyvalue0\" is not running: OBJECT_NOT_FOUND", 
"max_key": "ERROR:  Service unavailable: The object \"000033e6000030008000000000000000.postgresqlkeyvalue0\" is not running: OBJECT_NOT_FOUND", 
"min_key": "ERROR:  Service unavailable: The object \"000033e6000030008000000000000000.postgresqlkeyvalue0\" is not running: OBJECT_NOT_FOUND"
```

### Warning: Please confirm that this issue does not contain any sensitive information

- [X] I confirm this issue does not contain any sensitive information.

[DB-5778]: https://yugabyte.atlassian.net/browse/DB-5778?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by lingamsandeep** [comment]
@qvad - Do you have logs for this or a repro script to help with the repro ?

**Comment by lingamsandeep** [comment]
We believe this should have been fixed by d7cf0a74421676eb7271089bc186aabe0fdb7c85
